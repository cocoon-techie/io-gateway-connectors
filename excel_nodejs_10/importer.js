const xlsx = require("xlsx")

function importer(data) {
    let workbook = xlsx.read(data, {"type":"base64"});
    let name = workbook.SheetNames[0]
    let sheet = workbook.Sheets[name]
    let ranges = sheet['!ref'].split(":")
    let fromCol = ranges[0].charCodeAt(0)
    let toCol = ranges[1].charCodeAt(0)
    let fromRow = parseInt(ranges[0].substring(1))
    let toRow= parseInt(ranges[1].substring(1))
    let keys = []
    let res = []
    for(r = fromRow; r <=toRow ; r++) {
        let i = 0;
        rec = {}
        for(c=fromCol; c<=toCol; c++) {
            cell = String.fromCharCode(c)+r
            if(cell in sheet) {
                value = sheet[cell].v
                if(r==fromRow) {       
                    keys.push(value)       
                } else {
                    key = keys[i]
                    rec[key] = value
                } 
            }
            i++
        }
        if(r!=fromRow && Object.keys(rec).length >0) {
            res.push(rec)
        }
    }
    return res
}

module.exports = importer

