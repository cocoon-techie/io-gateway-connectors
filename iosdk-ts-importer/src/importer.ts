//const xlsx = require("xlsx")
import * as xlsx from "xlsx";

export function importer(data) {
  let workbook = xlsx.read(data, { type: "base64" });
  let name = workbook.SheetNames[0];
  let sheet = workbook.Sheets[name];
  let ranges = sheet["!ref"].split(":");
  let fromCol = ranges[0].charCodeAt(0);
  let toCol = ranges[1].charCodeAt(0);
  let fromRow = parseInt(ranges[0].substring(1));
  let toRow = parseInt(ranges[1].substring(1));
  let keys = [];
  let res = [];
  for (let r = fromRow; r <= toRow; r++) {
    let i = 0;
    let rec = {};
    for (let c = fromCol; c <= toCol; c++) {
      const cell = String.fromCharCode(c) + r;
      if (cell in sheet) {
        const value = sheet[cell].v;
        if (r == fromRow) {
          keys.push(value);
        } else {
          const key = keys[i];
          rec[key] = value;
        }
      }
      i++;
    }
    if (r != fromRow && Object.keys(rec).length > 0) {
      res.push(rec);
    }
  }
  return res;
}
