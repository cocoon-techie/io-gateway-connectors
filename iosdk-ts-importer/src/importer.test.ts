//const fs = require('fs')
import * as fs from "fs";
import { importer } from "./importer";

test("importer", () => {
  let data = fs.readFileSync("data/test.xlsx").toString("base64");
  let imp = importer(data);
  expect(imp).toMatchSnapshot();
});
