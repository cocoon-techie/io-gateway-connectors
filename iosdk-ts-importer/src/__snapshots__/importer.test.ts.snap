// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importer 1`] = `
Array [
  Object {
    "amount": 1,
    "due_date": 44197,
    "fiscal_code": "ISPXNB32R82Y766F",
    "invalid_after_due_date": false,
    "markdown": "# Welcome, Giovanni Rossi\\\\nThis is a test message.\\\\n.This message was sent using the IO-SDK.\\\\nYour fiscal code is ISPXNB32R82Y766F\\\\nWe hope you are enjoing using IO.\\\\n",
    "notice_number": 1,
    "subject": "Hi Giovanni",
  },
  Object {
    "amount": 1,
    "due_date": 44197,
    "fiscal_code": "ISPXNB32R82Y766D",
    "invalid_after_due_date": false,
    "markdown": "# Welcome, Luca Rossi\\\\nThis is a test message.\\\\n.This message was sent using the IO-SDK.\\\\nYour fiscal code is ISPXNB32R82Y766D\\\\nWe hope you are enjoing using IO.\\\\n",
    "notice_number": 1,
    "subject": "Hi Luca",
  },
]
`;
